builtin/nomad/platform.go

```diff
@@ -211,6 +211,18 @@ func (p *Platform) resourceJobCreate(
 	job.TaskGroups[0].Tasks[0].Config = config
 	job.TaskGroups[0].Tasks[0].Env = env
 
+	// Get Consul ACL token from environment
+	*job.ConsulToken, err = ConsulAuth()
```

Comment:
+1 to Brian's suggestion to test, I'm not sure if assigning it to the pointer would also assign it to the job object?